pkg/services/ngalert/state/state.go

```diff
@@ -28,6 +28,28 @@ type Evaluation struct {
 	EvaluationTime   time.Time
 	EvaluationState  eval.State
 	EvaluationString string
+	// Values contains the RefID and value of reduce and math expressions.
+	// It does not contain values for classic conditions as the values
+	// in classic conditions do not have a RefID.
+	Values map[string]EvaluationValue
+}
+
+// EvaluationValue contains the labels and value for a RefID in an evaluation.
+type EvaluationValue struct {
+	Labels data.Labels
+	Value  *float64
+}
+
+// NewEvaluationValues returns the labels and values for each RefID in the capture.
+func NewEvaluationValues(m map[string]eval.NumberValueCapture) map[string]EvaluationValue {
+	result := make(map[string]EvaluationValue)
```

Comment:
```suggestion
	result := make(map[string]EvaluationValue, len(m))
```

Apologies if I'm coming across as pedantic with the hints, they're quite important in this area given this is touch by very often by every alert _instance_.